test/File/FileInformationTest.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+declare(strict_types=1);
+
+namespace LaminasTest\Filter\File;
+
+use Laminas\Diactoros\UploadedFile;
+use Laminas\Filter\File\FileInformation;
+use PHPUnit\Framework\Attributes\CoversClass;
+use PHPUnit\Framework\TestCase;
+
+use function chmod;
+use function filesize;
+use function touch;
+use function unlink;
+
+use const UPLOAD_ERR_OK;
+
+/** @psalm-suppress InternalClass, InternalMethod, InternalProperty */
+#[CoversClass(FileInformation::class)]
+class FileInformationTest extends TestCase
+{
+    public function testThatANonExistentFileCannotBeCreatedFromAString(): void
+    {
+        $this->expectExceptionMessage('Cannot detect any file information');
+        FileInformation::factory('Foo');
+    }
+
+    public function testThatANonExistentFileCannotBeCreatedFromASapiFileArray(): void
+    {
+        $this->expectExceptionMessage('Cannot detect any file information');
+        FileInformation::factory([
+            'error'    => UPLOAD_ERR_OK,
+            'tmp_name' => 'Foo',
+            'name'     => 'Foo',
+            'type'     => 'text/plain',
+            'size'     => 0,
+        ]);
+    }
+
+    public function testExpectedValuesForFilePath(): void
+    {
+        $path = __DIR__ . '/fixtures/File1.txt';
+        $file = FileInformation::factory($path);
+
+        self::assertSame($path, $file->path);
+        self::assertNull($file->clientFileName);
+        self::assertNull($file->clientMediaType);
+        self::assertTrue($file->readable);
+        self::assertSame('File1.txt', $file->baseName);
+    }
+
+    public function testExpectedValuesForUploadedFile(): void
+    {
+        $path = __DIR__ . '/fixtures/File1.txt';
+
+        $upload = new UploadedFile(
+            $path,
+            filesize($path),
+            UPLOAD_ERR_OK,
+            'Foo.txt',
+            'text/plain',
+        );
+
+        $file = FileInformation::factory($upload);
```

Comment:
If you want to avoid the Laminas\Diactoros\UploadedFile import
```suggestion
                $mockStream = $this->createMock(StreamInterface::class);
        $mockStream->expects(self::once())->method('getMetadata')->with('uri')->willReturn($path);

        $mockFileUpload = $this->createMock(UploadedFileInterface::class);
        $mockFileUpload->expects(self::once())->method('getStream')->willReturn($mockStream);
        $mockFileUpload->expects(self::once())->method('getClientFilename')->willReturn('Foo.txt');
        $mockFileUpload->expects(self::once())->method('getClientMediaType')->willReturn('text/plain');
        
        $file = FileInformation::factory($mockFileUpload);
```

Reply:
The `UploadedFile` is simple and therefore I would not use a mock here.
And it's also much easier to read.

Reply:
That's fair. The mocking is little more complex. laminas/laminas-diactoros is only being imported for two tests, so this is just an alternative approach for if there's more value in not introducing a new dependency, but is a tradeoff. I'm not precious either way